app/models/locker_application.rb

```diff
@@ -58,9 +58,10 @@ def available_lockers_in_area_and_size(building:)
     context.order(:location)
   end
 
-  def self.search(uid:)
-    return where(complete: true) if uid.blank?
+  def self.search(uid:, archived:)
+    return joins(:user).where("users.uid = '#{uid}'").where('locker_applications.archived = false').where(complete: true) if uid.present?
```

Comment:
Would a single `where` be more readable here?  I could go either way.

```suggestion
    return joins(:user).where(['users.uid = ? and locker_applications.archived = false and complete = true', uid]) if uid.present?
```